components/Dialogs/src/MDCAlertController.m

```diff
@@ -186,6 +189,48 @@ - (void)addAction:(MDCAlertAction *)action {
   [self.view setNeedsLayout];
 }
 
+- (BOOL)mdc_adjustsFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+
+  // Update any action buttons to match the alert controller.
+  for (MDCButton *actionButton in self.actionButtons) {
+    actionButton.mdc_adjustsFontForContentSizeCategory = adjusts;
+  }
+
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    [self updateFontsForDynamicType];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(contentSizeCategoryDidChange:)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+}
+
+// Handles UIContentSizeCategoryDidChangeNotifications
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  [self updateFontsForDynamicType];
+}
+
+// Update the fonts used based on mdc_preferredFontForMaterialTextStyle and recalculate the
+// preferred content size.
+- (void)updateFontsForDynamicType {
+  self.titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleTitle];
```

Comment:
Lets see if we can use a delegation instead of inheritance. If a client would like to setup their own preferred fonts they would have to subclass and override our logic here.
How we could do it:
Instead of a bool that dictates behavior we could have a dynamicTypeDelegate property that mutates the various label's fonts of the MDCAlertController. What do you think?